### PR TITLE
Create RunWidget for Indirect/Inelastic interfaces

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR creates a RunWidget consisting of a View and a Presenter. It does not (yet) need a model.

The new RunWidget is used in the QENS Fitting interface. The plan is to start using it in all other Indirect/Inelastic interfaces. At that stage, it should be possible to start work on #37201 

Part of #37381

### To test:
https://developer.mantidproject.org/Testing/Inelastic/QENSFittingTests.html

The "Run" button should be disabled and say "Running..." when a tab is running

*This does not require release notes* because **it is an internal change**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by adriazalvarez — CHANGES_REQUESTED** [review]
When for some reason there is an error when clicking `Run`, the disabled `Running` button remains disabled. (Like open QENS fitting and click `Run` without loading anything)

**Comment by robertapplin** [comment]
> When for some reason there is an error when clicking `Run`, the disabled `Running` button remains disabled. (Like open QENS fitting and click `Run` without loading anything)

Thanks for spotting this, it should be working now

**Comment by robertapplin** [comment]
Converting to draft because I'm going to improve this by moving the validation to the run widget

**Comment by robertapplin** [comment]
Thanks for waiting, ready for another look @adriazalvarez

File: qt/scientific_interfaces/Inelastic/test/Common/Run/RunPresenterTest.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+
+#include "../MockObjects.h"
+#include "Common/Run/RunPresenter.h"
+#include "MantidQtWidgets/Common/MockUserInputValidator.h"
+
+#include <memory>
```
Comment (adriazalvarez) [inline qt/scientific_interfaces/Inelastic/test/Common/Run/RunPresenterTest.h:16]: is this include necessary?

**Review by adriazalvarez — CHANGES_REQUESTED** [review]
Thanks for the changes. 
The code looks neat and clear. I have tested with QENS Fitting interface and 
found no problem with the `Run` widget. 

I have added a small comment, and additionally I would suggest to change the name of the `Run` folder to maybe `Run Widget` ?

File: qt/scientific_interfaces/Inelastic/test/Common/Run/RunPresenterTest.h (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/test/Common/Run/RunPresenterTest.h:16]: I think I've included it as its used for `std::unique_ptr` and `std::make_unique`

**Review by adriazalvarez — APPROVED** [review]
Thanks for the changes.

**Pull request merged**